model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -287,7 +290,12 @@ private UserSessionEntity getUserSessionEntityFromPersistenceProvider(RealmModel
             return null;
         }
 
-        return importUserSession(realm, offline, persistentUserSession);
+        UserSessionEntity sessionEntity = importUserSession(realm, offline, persistentUserSession);
+        if (sessionEntity == null) {
+            persister.removeUserSession(sessionId, offline);
```

Comment:
Out of curiosity, why we need to delete the offline session if `importUserSession` didn't find in the DB? Is it because we want to remove associated offline client sessions?

Reply:
At this point we now the session is in the offline store but is expired. So it's removed now. If not it will be removed by some scheduled task later. I think that I needed this for the tests (because the sessions created by the new test remained in the store and other tests related with the persister counted them).